[Gardening] Use apostrophes around symbols (but not file names) in diagnostic messages

WowbaggersLiquidLunch opened the pull request:
Make almost all diagnostic messages use apostrophes around symbols. Most of them already do. This makes all the messages' (except those with file names) formatting consistent.

WowbaggersLiquidLunch commented:
@swift-ci please test

WowbaggersLiquidLunch commented:
@swift-ci please smoke test

xwu commented on include/swift/AST/DiagnosticsCommon.def:
```diff
@@ -145,7 +145,7 @@ NOTE(fixit_correct_source_location_file,none,
 
 // Usually, but not always, emitted from the driver
 ERROR(error_two_files_same_name,none,
-      "filename \"%0\" used twice: '%1' and '%2'",
```
To my understanding, the apostrophe is used for indicating code voice in diagnostics; I think it's reasonable for file names to retain their double quotation marks.

WowbaggersLiquidLunch replied:
Sorry for this late response!

In my opinion, the distinction between code voice and file names are not that clear, especially when it comes to file paths. However, I can see the point you might be making (paraphrasing it to make sure I understand correctly): a file name does not originate from a code context, so even if it used in code (or not, as in the case of this marked line) it should not be treated as code voice. Is this what you mean?

xwu replied:
Here's how I would format the same sentence when writing on GitHub:

> filename "foobarbaz.png" used twice

I wouldn't use code voice. That is, I wouldn't write...

> filename `foobarbaz.png` used twice

...nor would I write...

> filename `"foobarbaz.png"` used twice

For the purposes of Swift diagnostics, we use single quotes instead of the backtick for code voice, but I wouldn't use code voice here.

WowbaggersLiquidLunch replied:
I understand this part. 

But I question if it is indeed the case that file names should use double quotes. There are many places where file names are already using single quotes, including the diagnostic message on [line 35 of the same file](https://github.com/apple/swift/blob/e3fece033c8e764004abf40c381846af16bfcf88/include/swift/AST/DiagnosticsCommon.def#L39).

xwu replied:
🤷 In either case, it's fair to say that file names aren't "symbols" and if no consensus we can subset those changes out of this PR.

WowbaggersLiquidLunch replied:
Sounds reasonable to me. I will remove my changes to file names' quotations.

WowbaggersLiquidLunch replied:
done in f6fd93781670a1fe5ec2ebe3e8fb5107f43922cc

WowbaggersLiquidLunch commented:
@swift-ci please test

WowbaggersLiquidLunch commented:
@swift-ci please smoke test

WowbaggersLiquidLunch commented:
@swift-ci please smoke test

xwu commented on include/swift/AST/DiagnosticsDriver.def:
```diff
@@ -109,7 +109,7 @@ WARNING(warn_cannot_stat_input,none,
 
 WARNING(warn_unable_to_load_dependencies, none,
-        "unable to load dependencies file \"%0\", disabling incremental mode",
+        "unable to load dependencies file '%0', disabling incremental mode",
```
I think this is a file name too; not sure about the change above it but it also refers to a file map. The rest of the changes lgtm!

WowbaggersLiquidLunch replied:
I  think the one above it returns the `NAME` of one of these: https://github.com/apple/swift/blob/9370929518b0053e61f2bf94754410841091055e/include/swift/Basic/FileTypes.def, which is not a file name

I'll remove the highlighted change.

WowbaggersLiquidLunch replied:
done in cb507c514da88d7c10ed1d725a32a36bf831b619

WowbaggersLiquidLunch committed:
[gardening] use apostrophes around symbols (not file name) in diagnostic messages

Most diagnostic messages already use apostrophes around symbols. This commit changes some that used double quotation marks to use apostrophes too. Quotations around file names are unchanged.

WowbaggersLiquidLunch commented:
@swift-ci please smoke test
